src/util-mpm-ac.c

```diff
@@ -131,10 +153,10 @@ static void SCACGetConfig()
 static inline int SCACReallocState(SCACCtx *ctx, uint32_t cnt)
 {
     void *ptmp;
-    int size = 0;
+    size_t size = 0;
 
     /* reallocate space in the goto table to include a new state */
-    size = cnt * ctx->single_state_size;
+    size = SCACCheckSafeSizetMult((size_t) cnt, (size_t) ctx->single_state_size);
     ptmp = SCRealloc(ctx->goto_table, size);
```

Comment:
Interestingly, scan-build-3.9 now warns that size could be 0:
<pre>
util-mpm-ac.c:160:12: warning: Call to 'realloc' has an allocation size of 0 bytes
    ptmp = SCRealloc(ctx->goto_table, size);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./util-mem.h:193:14: note: expanded from macro 'SCRealloc'
    ptrmem = realloc((x), (a)); \
             ^~~~~~~~~~~~~~~~~
1 warning generated.
</pre>
I guess a simple size check would fix it. It is out goal to get to 0 issues in scan-build (and other checkers), even if they are sometimes far-fetched or even wrong. So we'll need a fix.